src/ConditionalApplyPlugin.groovy

```diff
@@ -13,6 +14,7 @@ public class ConditionalApplyPlugin implements TerraformEnvironmentStagePlugin {
     public void apply(TerraformEnvironmentStage stage) {
         stage.decorateAround(CONFIRM, onlyOnExpectedBranch())
         stage.decorateAround(APPLY, onlyOnExpectedBranch())
+        stage.decorateAround(DESTROY, onlyOnExpectedBranch())
     }
```

Comment:
If we only attack UseCase 1 in this PR (destroy-only), we no longer need these changes.